server/sonar-server/src/main/java/org/sonar/server/ce/ws/ActivityStatusAction.java

```diff
@@ -106,11 +105,11 @@ private ActivityStatusWsResponse doHandle(ActivityStatusWsRequest request) {
     return Optional.fromNullable(component);
   }
 
-  private void checkPermissions(@Nullable String componentUuid) {
-    if (componentUuid == null) {
-      userSession.checkPermission(GlobalPermissions.SYSTEM_ADMIN);
+  private void checkPermissions(Optional<ComponentDto> component) {
+    if (component.isPresent()) {
+      userSession.checkComponentPermission(UserRole.ADMIN, component.get());
     } else {
-      userSession.checkComponentUuidPermission(UserRole.ADMIN, componentUuid);
+      userSession.checkPermission(GlobalPermissions.SYSTEM_ADMIN);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "checkPermission"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)